scripts/install_backend_python_libs.py

```diff
@@ -42,6 +43,24 @@ def _normalize_python_library_name(library_name):
     Returns:
         str. A normalized library name that is all lowercase.
     """
+    # There are certain python libraries that have added support for other
+    # libraries. For example, 'google-api-core[grpc]' means that a variant of
+    # the 'google-api-core' package is installed that supports grpc. However,
+    # the import names, the package directory names, and the metadata directory
+    # names remain the same (google-api-core). These variant libraries are
+    # specified in the 'requirements.txt' file with the string,
+    # `library[sub-library]`. However, the installed version of this library
+    # is still considered by pkg_resources as having the default name `library`.
+    # This scenario causes this script to find incorrect mismatches. To solve
+    # this problem, we remove the special support package designation
+    # (e.g [grpc]) in the brackets when parsing the requirements file.
+    # NOTE: This does not cause ambiguities because there is no viable scenario
+    # where both the library and a variant of the library exist in the
+    # directory. Both the default version and the variant are imported in the
+    # same way (e.g import google.api.core) and if pip allowed both versions,
+    # then there would be ambiguities in the imports. For this reason, it is
+    # safe to disambiguate the names by removing the brackets.
+    library_name = re.sub('\[[^\]]+\]', '', library_name)
```

Comment:
Can you add tests for this? Despite the long, well-written comment above, we should have a forced check to prevent this code from unintentionally regressing, if possible.

Reply:
Done, added a backend test to assert the library names match the regex that we expect.